lib/files.ncl

```diff
@@ -0,0 +1,63 @@
+let nix = import "./nix-interop/nix.ncl" in
+{
+  File = {
+    target
+      | doc m%"
+        The file to write to.
+        If null, defaults to the attribute name of the file.
+      "%
+      | nix.derivation.NullOr String
+      | default
+      = null,
+    content
+      | doc m%"
+        The content of the file.
+      "%
+      | nix.derivation.NixString,
+    materialisation_method : [| 'Symlink, 'Copy |]
+      | doc m%"
+        How the file should be materialized on-disk.
+
+        Symlinking is generally preferred as it makes it easier to track where
+        the files are coming from, but their target only exists after a first
+        call to Organist, which might be undesirable.
+      "%
+      | default = 'Symlink,
+  },
+  Files = { _ : File },
+
+  regenerate_files | Files -> nix.derivation.Derivation = fun files_to_generate =>
+      let regnerate_one | String -> File -> nix.derivation.NixString = fun key file_descr =>
+          let file_content = file_descr.content in
+          let target = if file_descr.target == null then key else file_descr.target in
+          let copy_command =
+            match {
+              'Symlink => "ln -s",
+              'Copy => "cp",
+            }
+              file_descr.materialisation_method
+          in
+          let file_in_store = nix.builtins.to_file key file_content in
+          nix-s%"
+            (chmod +w %{target}; rm %{target}) || true
```

Comment:
```suggestion
            rm -f %{target}
```

`chmod` requires rights to modify the inode, but we only need to remove the file from the directory. Also, `rm -f` doesn't complain about access rights and missing files.

Reply:
Indeed, thanks